Update docs according to QUnit 2.0 changes

jayeff opened the pull request:
Updated testing documentation to use new hook names `beforeEach`/`afterEach` and replace global assertions with `assert` argument. This change should match the recent changes to the blueprints (see #3197) 

I was not sure how to apply the `assert` argument with the [test helper example](http://www.ember-cli.com/#single-helper-per-file). Pointers how to update this part are appreciated.

stefanpenner commented on _posts/2013-04-11-testing.md:
```diff
@@ -60,20 +60,20 @@ import { test } from 'ember-qunit';
 
-module('An Integration test', {
-  setup: function() {
+Qunit.module('An Integration test', {
```
is this required now?

jayeff replied:
It will be with QUnit 2.0 (http://qunitjs.com/upgrade-guide-2.x/). The blueprints now generate tests like this (#3197 was merged a few days ago)

stefanpenner replied:
ah, ok. Fine for now, but going forward we should shim qunit so we can `import { module } from 'qunit';`

rwjblue replied:
@stefanpenner - Yes, we could definitely do that.

jayeff replied:
I'm not familiar how to do this. I'm guessing that this shim should be added to `ember-qunit`? What would be the next step here?

rwjblue replied:
I added a little shim in https://github.com/rwjblue/ember-qunit/pull/132.

rwjblue replied:
Updating blueprints to use the shim in https://github.com/ember-cli/ember-cli/pull/3224.

jayeff commented:
I just updated this PR to use the shim as well. 

@rwjblue: I changed all `import { test } from 'ember-qunit'` in the documentation to `import { test } from 'qunit'`. Is this change correct?

rwjblue commented:
@jayeff - No, both are needed depending on use case. For unit style tests `import { test, moduleFor } from 'ember-qunit';` is correct, for acceptance tests `import { test, module } from 'qunit';` is correct. If `ember-qunit` was used before, it should still be used.

jayeff commented on _posts/2013-04-11-testing.md:
```diff
@@ -52,28 +52,28 @@ All Ember Apps come with built-in [ember test helpers](http://emberjs.com/guides
 {% highlight javascript linenos %}
 import Ember from "ember";
-import { test } from 'ember-qunit';
+import { module, test } from 'qunit';
```
@rwjblue updated again - thanks for the clarification. I hope this line is now as it should be? Was the previous `import { test } from 'ember-qunit'` incorrect or would both work?

rwjblue replied:
This update is correct...

jayeff commented:
Today I tried to update the single helper example and I somehow succeeded (see 55a046b). To use the `routesTo` helper you now have to pass it the assert argument as well like so `routesTo(assert, '/post', 'post.index')`.

It works but it's not elegant. I hope there's a way how this could be improved but I have no clue how.

rwjblue commented:
This needs a rebase, but is ready to land when you do...

jayeff commented:
Another PR with most of my changes was merged in the meantime. I updated this PR to include two of my changes which were not covered by the other PR #3256.

stefanpenner commented:
thanks :)
